optimum/onnxruntime/modeling_ort.py

```diff
@@ -297,59 +286,72 @@ def dtype(self) -> torch.dtype:
-        device).
-        """
-        return self._device
-
-    @device.setter
-    def device(self, **kwargs):
-        raise AttributeError("The device attribute is read-only, please use the `to` method to change the device.")
+    def providers(self):
+        # all providers
+        return self._providers
 
     @property
-    def use_io_binding(self):
-        return check_io_binding(self.providers, self._use_io_binding)
-
-    @use_io_binding.setter
-    def use_io_binding(self, value: bool):
-        self._use_io_binding = value
-
-    def to(self, device: Union[torch.device, str, int]):
-        """
-        Changes the ONNX Runtime provider according to the device.
-
-        Args:
-            device (`torch.device` or `str` or `int`):
-                Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, a positive will run
-                the model on the associated CUDA device id. You can pass native `torch.device` or a `str` too.
+    def provider(self):
+        # main provider
+        return self._providers[0]
 
-        Returns:
-            `ORTModel`: the model placed on the requested device.
-        """
+    @property
+    def providers_options(self):
+        # all provider options
+        return self._providers_options
 
-        device, provider_options = parse_device(device)
+    @property
+    def provider_options(self):
+        # main provider options
+        return self.providers_options[self.provider]
+
+    def to(self, *args, device: Optional[Union[torch.device, str, int]] = None, dtype: Optional[torch.dtype] = None):
+        for arg in args:
+            if isinstance(arg, torch.device):
+                device = arg
+            elif isinstance(arg, (int, str)):
+                device = torch.device(arg)
+            elif isinstance(arg, torch.dtype):
+                dtype = arg
+
+        if dtype is not None and dtype != self.dtype:
+            raise NotImplementedError(
+                f"Cannot change the dtype of the pipeline from {self.dtype} to {dtype}. "
+                f"Please export the pipeline with the desired dtype."
+            )
 
-        if device.type == "cuda" and self.providers[0] == "TensorrtExecutionProvider":
+        if device is None or device == self.device:
             return self
 
+        device, provider_options = parse_device(device)
         provider = get_provider_for_device(device)
-        validate_provider_availability(provider)  # raise error if the provider is not available
+        validate_provider_availability(provider)
 
-        # IOBinding is only supported for CPU and CUDA Execution Providers.
-        if device.type == "cuda" and self._use_io_binding is False and provider == "CUDAExecutionProvider":
+        if self.use_io_binding is False and provider == "CUDAExecutionProvider":
             self.use_io_binding = True
```

Comment:
This overrides use_io_binding choice from user. What if user want to run performance test with io binding disabled?

I suggest that:
    if use_io_binding is None: change it to True
    if not use_io_binding and it is cuda provider, log a warning.

Reply:
This is the default behavior in other ORTModels, I kept it for consistency (I'm not a fan of it)